src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -595,6 +614,264 @@ class D
             Diagnostic(ErrorCode.ERR_DuplicateInterceptor, @"InterceptsLocation(""Program.cs"", 3, 3)").WithLocation(14, 6));
     }
 
+    [Fact]
+    public void InterceptorVirtual_01()
+    {
+        // Intercept a method call with a call to a virtual method on the same type.
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            C c = new C();
+            c.M();
+
+            c = new D();
+            c.M();
+
+            class C
+            {
+                [Interceptable]
+                public void M() => throw null!;
+
+                [InterceptsLocation("Program.cs", 5, 3)]
+                [InterceptsLocation("Program.cs", 8, 3)]
+                public virtual void Interceptor() => Console.Write("C");
+            }
+
+            class D : C
+            {
+                public override void Interceptor() => Console.Write("D");
+            }
+
+            namespace System.Runtime.CompilerServices
+            {
+                [AttributeUsage(AttributeTargets.Method)]
+                public sealed class InterceptableAttribute : Attribute { }
+
+                [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+                public sealed class InterceptsLocationAttribute : Attribute
+                {
+                    public InterceptsLocationAttribute(string filePath, int line, int character)
+                    {
+                    }
+                }
+            }
+            """;
+
+        var verifier = CompileAndVerify((source, "Program.cs"), expectedOutput: "CD");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void InterceptorVirtual_02()
+    {
+        // Intercept a call with a virtual method call on the base type.
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            D d = new D();
+            d.M();
+
+            class C
+            {
+                [InterceptsLocation("Program.cs", 5, 3)]
+                public virtual void Interceptor() => throw null!;
```

Comment:
Spec: Makes me wonder, what's the benefit of allowing `InterceptsLocation` on instance methods?

Reply:
I think you could say that the feature only works with static methods and not lose any expressivity. I could imagine, though, a generator adding instance interceptors to a partial type, to intercept calls to other methods in the type.

Reply:
Added a note to test plan for spec question